internal/pkg/archiver/walker.go

```diff
@@ -19,67 +17,74 @@ type FileItem struct {
 	RelPath  string
 	FileInfo os.FileInfo
 	Link     string
+	Error    error
+}
+
+type walkerOptions struct {
+	skipRoot        bool
+	maxRecurseDepth int
+}
+
+// WalkerOption configures Walker.
+type WalkerOption func(*walkerOptions)
+
+// SkipRoot skips root path if it's a directory.
+func SkipRoot() WalkerOption {
```

Comment:
There is one convention I _really_ like with functional option func naming, and that is to prefix the funds with `With`. Sin this case it would be `WithSkipRoot` and `MaxRecurseDepth` is `WithMaxRecurseDepth`. It helps the readability of the code and makes it clear that it is a functional option. Just my two cents.

Reply:
thanks, updated!